boefjes/plugins/kat_nmap/schema.json

```diff
@@ -0,0 +1,20 @@
+{
```

Comment:
Since it's a JSON schema, I think it would be very nice to have more descriptive properties. For example:
- define a range for `TOP_PORTS` (using `minimum` and `maximum` keywords, or `exclusiveMinimum` and `exclusiveMaximum` for expressing exclusive range)
- define an enumeration for `PROTOCOL` using `enum`
- define default values using `default`

Reply:
Agreed, but in my opinion that would be a feature request for Rocky (see also the discussion on https://github.com/minvws/nl-kat-rocky/pull/91)

Reply:
It would be both. One for setting those limits in this schema, and the second to make sure that we can actually render those limits and types in all their glory in the frontend.

Reply:
@ammar92 @underdarknl The settings are saved as key-value `VARCHAR`'s at this point in the Katalogus. Supporting numeric validators on fields that we retrieve from the database as strings is not very well supported and would require manual type coercion. I don't think that's worth the time.

I would opt for enforcing settings to be strings and do as much validation as you like with string validators (a regex is possible as well for instance). Any additional check can always be performed in the boefje. This makes the most sense in my opinion when these settings should act as a replacement for environment variables. Any complex object should either be an OOI or saved as a json string.

Other options include changing the type of `value` in the Katalogus database to `JSON`, in which case we can save all settings in one database entry that supports all primitive and compound json types. Or doing manual type coercion for all json schema types, but I think that's the worst of both worlds.

Reply:
Yes, we should definitely move to json in the katalogus, and json schema's for the expected structure of those json blobs.
Extracting the various parameters, env vars etc should then be handled in the runtime. This is a new feature though.

Reply:
@underdarknl https://github.com/minvws/nl-kat-boefjes/issues/59

But I think integers will not be properly supported in the current Katalogus setup, so I would opt for using strings for the time being.

Reply:
Since according to my unit tests, we cannot enable nmap now.